gateway/api/user/user.go

```diff
@@ -218,29 +276,40 @@ func Update(c *gin.Context) {
 //	@Router			/users [get]
 func List(c *gin.Context) {
 	ctx := storagev2.ParseContext(c)
-	users, err := pgusers.New().FetchAll(ctx)
+	users, err := models.ListUsers(ctx.OrgID)
 	if err != nil {
 		log.Errorf("failed listing users, err=%v", err)
 		sentry.CaptureException(err)
 		c.JSON(http.StatusInternalServerError, gin.H{"message": "failed listing users"})
 		return
 	}
-	userList := []openapi.User{}
-	for _, u := range users {
-		userList = append(userList,
+
+	usersList := []openapi.User{}
+	for i, u := range users {
+		userGroups, err := models.GetUserGroupsByUserID(users[i].ID)
```

Comment:
This is very bad, because you're executing an statement for each user record.
You could improve this by:
- [good] Performing one query to select all groups from an organization and them build the list payload
- [better] Perform a single query that return the groups in a single object

Reply:
indeed it is, but how we designed our user groups right now is far from ideal, so the controller layer will have to pay this price during this refactor until we match the simplicity we want